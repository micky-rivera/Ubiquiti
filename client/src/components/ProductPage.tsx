import React from 'react';
import { useAppSelector } from '../hooks/hooks';

function ProductPage() {
  const state = useAppSelector(state => state.app);

  const resolution = '257x257';

  const imageUrl = `https://static.ui.com/fingerprint/ui/icons/${state.chosenProduct.deviceId}_${resolution}.png`;

  return (
    <>
      <div className='product-page'>
          <img className='product-page__img' src={imageUrl} />
          <div className='product-page__row'>
                  <p className='product-page__label'>Name</p>
                  <p className='product-page__content'>{state.chosenProduct.name}</p>
            </div>
          <div className='product-page__row'>
                  <p className='product-page__label'>Product line</p>
                  <p className='product-page__content'>{state.chosenProduct.line}</p>
            </div>
          {state.chosenProduct.details.map((detail, index) => (
              <div key={index} className='product-page__row'>
                  <p className='product-page__label'>{detail.label}</p>
                  <p className='product-page__content'>{detail.content}</p>
              </div>
          ))}
      </div>
    </>
  );
}

export default ProductPage;
