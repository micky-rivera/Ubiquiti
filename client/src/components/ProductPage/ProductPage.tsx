import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAppSelector } from '../../hooks/hooks';

function ProductPage() {
  const state = useAppSelector(state => state.app);
  const navigate = useNavigate();

  const resolution = '257x257';

  const imageUrl = `https://static.ui.com/fingerprint/ui/icons/${state.chosenProduct.deviceId}_${resolution}.png`;

  const handleClick = () => {
    navigate('/');
  }

  useEffect(() => {
    if (state.chosenProduct.name === '') {
        navigate('/');
    }
  }, [])

  return (
    <>
        <div className='product-page__nav'>
            <div className='product-page__back-btn'>
            <svg data-testid='back-btn-svg' onClick={handleClick} className='product-page__back-btn-svg' width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M12.6516 15.253C12.8471 15.0575 12.8471 14.7405 12.6516 14.545L8.40863 10.303L12.6516 6.06098C12.8471 5.86553 12.847 5.54857 12.6514 5.35323C12.456 5.15811 12.1394 5.15822 11.9441 5.35348L7.70163 9.596C7.31117 9.98646 7.31171 10.6201 7.70218 11.0105L11.9437 15.253C12.1392 15.4485 12.4561 15.4485 12.6516 15.253Z" fill="#838691"/>
            </svg>
            </div>
            <p className='product-page__title' data-testid='title'>{state.chosenProduct.name}</p>
        </div>
        <div className='product-page'>
            <div className='product-page__img-container'>
                <img className='product-page__img' src={imageUrl} data-testid='img' />
            </div>
            <div className='product-page__info' data-testid='info'>
                <div className='product-page__row'>
                        <p className='product-page__label'>Name</p>
                        <p className='product-page__content'>{state.chosenProduct.name}</p>
                    </div>
                <div className='product-page__row'>
                        <p className='product-page__label'>Product line</p>
                        <p className='product-page__content'>{state.chosenProduct.line}</p>
                    </div>
                {state.chosenProduct.details.map((detail, index) => (
                    <div key={index} className='product-page__row'>
                        <p className='product-page__label'>{detail.label}</p>
                        <p className='product-page__content'>{detail.content}</p>
                    </div>
                ))}
            </div>
        </div>
    </>
  );
}

export default ProductPage;
