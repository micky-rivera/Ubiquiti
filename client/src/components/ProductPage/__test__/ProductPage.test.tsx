import React from "react";
import * as ReactDOM from "react-dom";
import ProductPage from '../ProductPage';
import { Provider } from 'react-redux'
import {store} from '../../../store/store'
import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>, div)
})

it('renders back button svg', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>);
    expect(getByTestId('back-btn-svg')).toBeInTheDocument();
})

it('renders title', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>);
    expect(getByTestId('title')).toBeInTheDocument();
})

it('renders image', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>);
    expect(getByTestId('img')).toBeInTheDocument();
})

it('renders information table', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>);
    expect(getByTestId('info')).toBeInTheDocument();
})

it('matches snapshot', () => {
    const tree = TestRenderer.create(<BrowserRouter><Provider store={store}><ProductPage /></Provider></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})