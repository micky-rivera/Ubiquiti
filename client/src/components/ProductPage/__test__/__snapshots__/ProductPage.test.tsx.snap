// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="product-page__nav"
  >
    <div
      className="product-page__back-btn"
    >
      <svg
        className="product-page__back-btn-svg"
        data-testid="back-btn-svg"
        fill="none"
        height="20"
        onClick={[Function]}
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.6516 15.253C12.8471 15.0575 12.8471 14.7405 12.6516 14.545L8.40863 10.303L12.6516 6.06098C12.8471 5.86553 12.847 5.54857 12.6514 5.35323C12.456 5.15811 12.1394 5.15822 11.9441 5.35348L7.70163 9.596C7.31117 9.98646 7.31171 10.6201 7.70218 11.0105L11.9437 15.253C12.1392 15.4485 12.4561 15.4485 12.6516 15.253Z"
          fill="#838691"
        />
      </svg>
    </div>
    <p
      className="product-page__title"
      data-testid="title"
    >
      Access Point Wifi Something idk
    </p>
  </div>,
  <div
    className="product-page"
  >
    <div
      className="product-page__img-container"
    >
      <img
        className="product-page__img"
        data-testid="img"
        src="https://static.ui.com/fingerprint/ui/icons/06a25b40-ef1f-463a-82d9-13236866ea3d_257x257.png"
      />
    </div>
    <div
      className="product-page__info"
      data-testid="info"
    >
      <div
        className="product-page__row"
      >
        <p
          className="product-page__label"
        >
          Name
        </p>
        <p
          className="product-page__content"
        >
          Access Point Wifi Something idk
        </p>
      </div>
      <div
        className="product-page__row"
      >
        <p
          className="product-page__label"
        >
          Product line
        </p>
        <p
          className="product-page__content"
        >
          Ubiquiti Product
        </p>
      </div>
      <div
        className="product-page__row"
      >
        <p
          className="product-page__label"
        >
          price
        </p>
        <p
          className="product-page__content"
        >
          astronomic
        </p>
      </div>
    </div>
  </div>,
]
`;
